src/Services/ApmAgentService.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace AG\ElasticApmLaravel\Services;
+
+use AG\ElasticApmLaravel\Agent;
+use Illuminate\Foundation\Application;
+use Nipwaayoni\Events\Transaction;
+use Psr\Http\Message\RequestInterface;
+
+class ApmAgentService
+{
+    /**
+     * @var Application
+     */
+    protected $app;
+
+    public function __construct(Application $app)
```

Comment:
Will not injecting the Agent work instead?

Reply:
I modeled this on the existing ApmCollectorService, which injects some dependencies, but use `App::make()` to get the Agent. It's easy enough to change. Do you want me to also change the ApmCollectorService at the same time?

Reply:
The `ApmCollectorService` class still needs access to the app to create the instances of the collectors, but we could also inject the Agent directly. I think it's better to inject the Agent, it makes the dependencies of the class very clear.

Reply:
I've updated both service classes to inject the Agent and addressed that change in the tests. The unit tests and a quick real-world test look good. This should be ready for your final consideration for merge.